### ISSUE TITLE
Persistent attachments in Mailable

### ISSUE THREAD

**Issue by JBtje** [laravel/framework#30857]
- Laravel Version: 5.8.* & 6.7.*
- PHP Version: 7.3.10
- Database Driver & Version: MariaDB 10.4.8

### Description:
If you run `render()` prior to sending the email, it will double the attachments.

### Steps To Reproduce:

```php
namespace App\Mail;

use Illuminate\Bus\Queueable;
use Illuminate\Mail\Mailable;
use Illuminate\Queue\SerializesModels;
use Illuminate\Support\Facades\Storage;

class RegisterReminder extends Mailable
{
    use Queueable, SerializesModels;

    public function __construct() {}

    public function build()
    {
        return $this->from( [
            'address' => config( 'mail.from.address' ),
            'name'    => config( 'mail.from.name' ),
        ] )
                    ->view( 'mail.patient.reminder' )
                    ->subject( 'Some reminder with attachments' )
                    ->attach( Storage::disk( 'resources' )->path( 'pdf-with-something.pdf' ), [
                        'as'   => 'PDF With Something.pdf',
                        'mime' => 'application/pdf',
                    ] );
    }
}
```

```php
use App\Mail\RegisterReminder;
use Illuminate\Support\Facades\Mail;

// Generate new email
$mailTemplate = new RegisterReminder();

// Obtain the content of the mail, for other purposes.
$mailContent = $mailTemplate->render();

// Add mail to the queue
Mail::to( 'my@email.com' )
    ->queue( $mailTemplate->onQueue( 'emails' ) );
```
#### Result:
One email with 2 attachments (both same name / content)

#### Expected:
One email, with one attachment

However, if you comment out `$mailContent = $mailTemplate->render();`, you'll receive the same email but with only one attachment.

Note: it does not matter if I send the mail via a database queue, or send the mail directly. Both methods provide two attachments if the email is rendered before sending, and one attachment if it is not rendered.
Note: `diskAttachments`, `rawAttachments`, `callbacks` will most likely behave the same as `attachments`.

I think it would make sense if these array variables are cleared in e.g. `render()`

**Comment by driesvints** [comment]
This is because both the `render` and `send` method of the Mailable will call `Container::getInstance()->call([$this, 'build']);`. So when you first render the mailable and later send it it'll call it twice and everything that happens inside the `build` method will happen twice. There can only be one from, one view and one subject so those don't get duplicated, just overwritten. But the `attach` method will attach the pdf again a second time and thus it'll appear twice when sending.

First question: is there anything preventing you from calling `$mailContent = $mailTemplate->render();` **after** the mailable's been sent? In that case you won't get the issue I suspect.

Second consideration is wether this is a bug at all. We could clear out all things set in the `build` method before calling `Container::getInstance()->call([$this, 'build']);` in `send` and `render` but that would be a little unpractical and perhaps even unwanted in some occasions. 

One other thing you can do is to reset the attachment array in the `build` method. I think this will be the easiest solution for now.

```php
    public function build()
    {
        $this->attachments = [];

        return $this->from( [
                'address' => config( 'mail.from.address' ),
                'name'    => config( 'mail.from.name' ),
            ] )
            ->view( 'mail.patient.reminder' )
            ->subject( 'Some reminder with attachments' )
            ->attach( Storage::disk( 'resources' )->path( 'pdf-with-something.pdf' ), [
                'as' => 'PDF With Something.pdf',
                'mime' => 'application/pdf',
            ] );
    }
```

**Comment by JBtje** [comment]
Resetting the `attachments` in the build was the solution I went with. My reason for running `render` prior to `send` ceased to exist when I switched to queuing the mails, but would also work.

However, I can't think of any reason why someone would want to have the same attachment twice in the same email; it is unexpected behaviour. Thus I stick with it being a bug and I think resetting all (array) variables prior to rendering is the best option.

I expected the below example to increase the amount of attachments on each run, however it did not...
```php
$mt = new RegisterReminder();
for( $i=1; $i<10; $i++ ) {
    Mail::to( User::find( $i )->email )->queue( $mt->onQueue( 'emails' ) );
}
```
Instead, it sent the first mail to user 1, the second email to both user 1 and 2, and the 3rd to user 1,2 and 3, etc.

Which leads to the question: how idiot proof does it need to be?
